apps/blockchain/lib/blockchain/interface/account_interface/cache.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Blockchain.Interface.AccountInterface.Cache do
+  alias Blockchain.Account.Address
+  defstruct cache: %{}
+
+  @type key_cache() :: %{
+          integer() => %{current_value: integer() | :deleted, initial_value: integer() | nil}
+        }
+  @type account_cache :: %{Address.t() => key_cache()}
+  @type t :: %__MODULE__{
+          cache: account_cache()
+        }
+
+  @spec get_current_value(t(), Address.t(), integer()) :: integer() | nil
+  def get_current_value(cache_struct, address, key) do
+    get_key_value(cache_struct.cache, address, key, :current_value)
+  end
+
+  @spec get_initial_value(t(), Address.t(), integer()) :: integer() | nil
+  def get_initial_value(cache_struct, address, key) do
```

Comment:
This is stylistic, so feel free to ignore, but what do you think of renaming these two functions that start with `get_*` into the same names without the `get_*`? 

```
Cache.get_current_value -> Cache.current_value

Cache.get_initial_value -> Cache.initial_value
```